homeassistant/components/ping/device_tracker.py

```diff
@@ -84,37 +89,20 @@ async def async_setup_entry(
     async_add_entities([PingDeviceTracker(entry, data.coordinators[entry.entry_id])])
 
 
-class PingDeviceTracker(CoordinatorEntity[PingUpdateCoordinator], ScannerEntity):
+class PingDeviceTracker(BasePingEntity, BaseTrackerEntity):
     """Representation of a Ping device tracker."""
 
-    def __init__(
-        self, config_entry: ConfigEntry, coordinator: PingUpdateCoordinator
-    ) -> None:
-        """Initialize the Ping device tracker."""
-        super().__init__(coordinator)
-
-        self._attr_name = config_entry.title
-        self.config_entry = config_entry
-
-    @property
-    def ip_address(self) -> str:
-        """Return the primary ip address of the device."""
-        return self.coordinator.data.ip_address
-
-    @property
-    def unique_id(self) -> str:
-        """Return a unique ID."""
-        return self.config_entry.entry_id
-
     @property
     def source_type(self) -> SourceType:
         """Return the source type which is router."""
         return SourceType.ROUTER
 
     @property
-    def is_connected(self) -> bool:
-        """Return true if ping returns is_alive."""
-        return self.coordinator.data.is_alive
+    def state(self) -> str:
```

Comment:
Overriding `state` isn't really according to our design intentions. I think this needs further discussion. I suggest we revert this PR for now.

Reply:
Revert PR https://github.com/home-assistant/core/pull/104682

Reply:
Didn't know that, I just saw that `ibeacon` and `private_ble_device` do it the same way (extend `BaseTrackerEntity` and override `state`) and thought it would be fine.

Reply:
The two device tracker entity classes we've documented are `ScannerEntity` and `TrackerEntity`. If those aren't sufficient to cover all cases we should make an architecture discussion and figure out what else is needed.

https://developers.home-assistant.io/docs/core/entity/device-tracker